pkg/hostagent/hostagent.go

```diff
@@ -65,6 +66,25 @@ type options struct {
 
 type Opt func(*options) error
 
+// Default nofile limit is too low on some system.
+// For example in the macOS standard terminal is 256.
+// It means that there are only ~240 connections available from the host to the vm.
+// That function increases the nofile limit for child processes, especially the ssh process
+//
+// More about limits in go: https://go.dev/issue/46279
+func init() {
```

Comment:
Not a huge fun of `init()`.

This can be just a regular function and called from `cmd/limactl/hostagent.go`